_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,419 @@
+```bash
+doctl projects list
+doctl projects resources list <project-id>
+```
+
+## Secure web traffic with HTTPS
+
+As the gateway service acts as the application front-end, in the k8s descriptors it is defined as a `LoadBalancer` service type. This exposes the service externally using the cloud provider's load balancer. DigitalOcean load balancers are a fully-managed, highly available network load balancing service. Load balancers distribute traffic to groups of droplets, which decouples the overall health of a backend service from the health of a single server to ensure that your services stay online.
+
+The standard practice is to secure web traffic to your application with HTTPs. For traffic encryption, you need a TLS (SSL) certificate. DigitalOcean also provides automatic certificate creation and renewal if you manage your domain with DigitalOcean's DNS, which is free. But domain registration is not provided.  To use DigitalOcean's DNS, you need to register a domain name with a registrar and update your domain's NS records to point to DigitalOcean's name servers.
+
+Then, for using DigitalOcean's managed domain and certificate, you must [delegate the domain](https://docs.digitalocean.com/tutorials/dns-registrars/), updating NS records in the registrar.
+
+**IMPORTANT NOTE**: Before changing the registrar NS records (the nameservers), add your domain to DigitalOcean, to minimize service disruptions.
+
+You can create the certificate and the domain at the same time, in the DigitalOcean control panel, when you set up the load balancer HTTPs forwarding.
+
+DigitalOcean load balancers support two main configurations for encrypted web traffic:
+
+- SSL termination: decrypts SSL requests at the load balancer and sends them unencrypted to the backend at the Droplets' private IP address. The slower and CPU-intensive work of decryption is performed at the load balancer, and certificate management is simplified. The traffic between the load balancer and the backend is secured by routing over the VPC network, but data is readable inside the private network.
+- SSL passthrough: sends the encrypted SSL requests directly to the backend at the Droplets' private IP address, traffic between the load balancer and the backend is secured. Every backend server must have the certificate, and client information contained in `X-forwarded-*` headers might be lost.
+
+Taking advantage of the simplified certificate management, in the following steps SSL termination will be configured, through the DigitalOcean control panel.
+
+Login to your DigitalOcean account, and in the left menu choose **Kubernetes**. Then choose your cluster, and on the cluster page, choose **Resources**. In the _LOAD BALANCERS_ list, choose the single load balancer that must have been created. On the load balancer page, choose the **Settings** tab. Click **Edit** in the _Forwarding Rules_. Add a forwarding rule for HTTPS in port 443, and in the certificate drop-down, choose **New certificate**.
```

Comment:
How do I log in? Do I just go to https://www.digitalocean.com/ and log in? This worked, but maybe we should link to `https://cloud.digitalocean.com/projects` here?

Reply:
For the forwarding rule, what are the protocol and port settings? Should I add a rule or replace the existing one?

<img width="928" alt="Screen Shot 2022-05-18 at 11 45 17" src="https://user-images.githubusercontent.com/17892/169031390-14ff428e-bc6f-4fce-8d33-18c3f9ac473c.png">

Reply:
I don't see a certificate drop-down. Where should I see this?

Reply:
I think if you set in the left HTTPS and 443 the drop down should be enabled.

Reply:
Yep. I see it now. Is it possible to configure HTTPS with something like https://nip.io like I do in my [Kubernetes to the Cloud post](https://developer.okta.com/blog/2021/06/01/kubernetes-spring-boot-jhipster#add-https-to-your-reactive-gateway)? I found that easier than using one of my custom domain names. If not, maybe we should mention it?

Reply:
I changed the DO link, to send the reader to the 60 day trial. I added a comment on why nip.io cannot be used.